parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -629,6 +629,23 @@ private void waitForClose(Duration timeout) throws TimeoutException, ExecutionEx
     }
 
     private void doClose(Duration timeout) throws TimeoutException, ExecutionException, InterruptedException {
+        try {
```

Comment:
could you please add a comment referencing github issue - and maybe a really short explanation for all the try / catch - that this is due to missing timeout / controlled shutdown on the poller / commit subsystems.

Reply:
added comment